### DISCUSSION TITLE
WP-CLI for site owners and admins

### DISCUSSION THREAD

**Discussion by zzap** [WordPress/developer-blog-content#243] [Topic Ideas]
WP-CLI is still scary to many developers, which is somewhat understandable given that it doesn't play nicely with Windows machines. A lot of developer tools, initially CLIs, have their GUIs today, and people opt to use them (which increases the fear of terminal).

So let's skip the setup part altogether and take a look at WP-CLI possibilities out of the box, i.e., hosting servers. The WP-CLI handbook already has a list of [hosting providers with preinstalled WP-CLI](https://make.wordpress.org/cli/handbook/references/hosting-companies/) and [plugins with their custom WP-CLI commands](https://make.wordpress.org/cli/handbook/references/tools/), so we're clear on "not recommending any company or product". 

We could go in general here and explain how WP-CLI can be used on the hosting provider's server, provide use cases, etc. If there were fewer providers, it would be beneficial to have a little walkthrough of the dashboard and show where the terminal is and how to start it, but each of them certainly has a guide on their blog, so we could just say, "Find the blog post and follow instructions".

I'd focus here mainly on site owners and administrators and what they can do in specific situations, e.g., manage user access (when the site is hacked or similar), install plugins for debugging and running various scans, switch to the default theme and deactivate all plugins to find the error, move a large number of posts to a different category/tag as a recommendation from an SEO agency, etc.  

Goal: empower experienced users to be more advanced; support hosting providers with WP-CLI; support plugin authors who include custom WP-CLI commands; promote the WP-CLI tool as the most powerful and quickest tool for managing a WordPress instance.

The title is debatable. I don't like it, but it describes what the article is about.

**Comment by zzap** [comment]
I also have an objection for this topic because we're running `dev blog` not `user blog` so maybe worth discussing.

**Reply by justintadlock** [reply]
"Site owners and administrators" in this case feels like "developers" to me. Or that other group that we cater to called "site builders." Either way, it feels like a good topic for the Dev Blog.

**Comment by abhansnuk** [comment]
> WP-CLI is still scary to many developers, which is somewhat understandable given that it doesn't play nicely with Windows machines. A lot of developer tools, initially CLIs, have their GUIs today, and people opt to use them (which increases the fear of terminal).
> 
> So let's skip the setup part altogether and take a look at WP-CLI possibilities out of the box, i.e., hosting servers. The WP-CLI handbook already has a list of [hosting providers with preinstalled WP-CLI](https://make.wordpress.org/cli/handbook/references/hosting-companies/) and [plugins with their custom WP-CLI commands](https://make.wordpress.org/cli/handbook/references/tools/), so we're clear on "not recommending any company or product".
> 
> We could go in general here and explain how WP-CLI can be used on the hosting provider's server, provide use cases, etc. If there were fewer providers, it would be beneficial to have a little walkthrough of the dashboard and show where the terminal is and how to start it, but each of them certainly has a guide on their blog, so we could just say, "Find the blog post and follow instructions".
> 
> I'd focus here mainly on site owners and administrators and what they can do in specific situations, e.g., manage user access (when the site is hacked or similar), install plugins for debugging and running various scans, switch to the default theme and deactivate all plugins to find the error, move a large number of posts to a different category/tag as a recommendation from an SEO agency, etc.
> 
> Goal: empower experienced users to be more advanced; support hosting providers with WP-CLI; support plugin authors who include custom WP-CLI commands; promote the WP-CLI tool as the most powerful and quickest tool for managing a WordPress instance.
> 
> The title is debatable. I don't like it, but it describes what the article is about.

Fully support these ideas @zzap and that it would be a practical and useful blog, also have a longer life. It could cross-link to any relevant materials on Learn L, docs, core too. I remember a documented resource on WP-CLI which Alain shared for a couple of contributor days I have been involved with too. It may be published in the handbook.

**Comment by ironnysh** [comment]
> WP-CLI is still scary to many developers

@zzap, this popped in my RSS today: https://wpcli.dev.
I'd say that it's a must-have article :-)

**Comment by bph** [comment]
The topic was approved and conversation moves to the issue

- #248

**Comment by zzap** [comment]
Published https://developer.wordpress.org/news/2024/09/09/website-security-checks-wp-cli-for-site-owners-and-administrators/
